tests/0081-admin.c

```diff
@@ -1131,6 +1131,419 @@ static void do_test_unclean_destroy (rd_kafka_type_t cltype, int with_mainq) {
+        test_timing_t timing;
+        rd_kafka_resp_err_t exp_err = RD_KAFKA_RESP_ERR_NO_ERROR;
+
+        TEST_SAY(_C_MAG "[ %s DeleteRecords with %s, op_timeout %d ]\n",
+                 rd_kafka_name(rk), what, op_timeout);
+
+        if (op_timeout != -1) {
+                options = rd_kafka_AdminOptions_new(
+                        rk, RD_KAFKA_ADMIN_OP_ANY);
+
+                err = rd_kafka_AdminOptions_set_operation_timeout(
+                        options, op_timeout, errstr, sizeof(errstr));
+                TEST_ASSERT(!err, "%s", rd_kafka_err2str(err));
+        }
+
+
+        for (i = 0 ; i < MY_DEL_RECORDS_CNT ; i++) {
+                char *topic = rd_strdup(test_mk_topic_name(__FUNCTION__, 1));
+
+                topics[i] = topic;
+                exp_mdtopics[exp_mdtopic_cnt++].topic = topic;
+        }
+
+        /* Create the topics first. */
+        test_CreateTopics_simple(rk, NULL, topics,
+                                 MY_DEL_RECORDS_CNT,
+                                 partitions_cnt /*num_partitions*/,
+                                 NULL);
+
+        /* Verify that topics are reported by metadata */
+        test_wait_metadata_update(rk,
+                                  exp_mdtopics, exp_mdtopic_cnt,
+                                  NULL, 0,
+                                  15*1000);
+
+        /* Produce 100 msgs / partition */
+        for (i = 0 ; i < MY_DEL_RECORDS_CNT; i++ ) {
+                int32_t partition;
+                for (partition = 0 ; partition < partitions_cnt; partition++ ) {
+                        test_produce_msgs_easy(topics[i], 0, partition, msgs_cnt);
+                }
+        }
+
+        del_records = rd_kafka_topic_partition_list_new(10);
+
+        /* Wipe all data from topic 0 */
+        for (i = 0 ; i < partitions_cnt; i++)
+                rd_kafka_topic_partition_list_add(del_records, topics[0], i)->offset = RD_KAFKA_OFFSET_END;
+
+        /* Wipe all data from partition 0 in topic 1 */
+        rd_kafka_topic_partition_list_add(del_records, topics[1], 0)->offset = RD_KAFKA_OFFSET_END;
+
+        /* Wipe some data from partition 2 in topic 1 */
+        rd_kafka_topic_partition_list_add(del_records, topics[1], 2)->offset = msgs_cnt / 2;
+
+        /* Not changing the offset (out of range) for topic 2 partition 0 */
+        rd_kafka_topic_partition_list_add(del_records, topics[2], 0);
+
+        /* Offset out of range for topic 2 partition 1 */
+        rd_kafka_topic_partition_list_add(del_records, topics[2], 1)->offset = msgs_cnt + 1;
+
+        TIMING_START(&timing, "DeleteRecords");
+        TEST_SAY("Call DeleteRecords\n");
+        err = rd_kafka_DeleteRecords(rk, del_records, options, q, tmout_multip(20*1000));
+        TEST_ASSERT(!err, "%s", rd_kafka_err2str(err));
+        TIMING_ASSERT_LATER(&timing, 0, 1000);
+
+        TIMING_START(&timing, "DeleteRecords.queue_poll");
+
+        const rd_kafka_DeleteRecords_result_t *res;
```

Comment:
All variable defs need to go at head of scope to be compliant with old compilers